packages/core/admin/ee/server/services/review-workflows/validation.js

```diff
@@ -32,6 +33,11 @@ module.exports = ({ strapi }) => {
       if (stages.length > this.limits.stagesPerWorkflow) {
         throw new ValidationError(ERRORS.STAGES_LIMIT);
       }
+      // Validate stage names are not duplicated
+      const stageNames = stages.map((stage) => stage.name);
+      if (uniq(stageNames).length !== stageNames.length) {
+        throw new ValidationError(ERRORS.DUPLICATED_STAGE_NAME);
```

Comment:
Actually: could you make this a yup validation and pass on the error message? This would be helpful, because then the FE already can render the error messages properly.

Reply:
I got something working, if 3 stages have the same name backend will return the following error:

```
{
    "status": 400,
    "name": "ValidationError",
    "message": "3 errors occurred",
    "details": {
        "errors": [
            {
                "path": [ "stages", "0", "name"],
                "message": "Stage must be unique",
                "name": "ValidationError"
            },
            {
                "path": [ "stages", "1", "name"],
                "message": "Stage must be unique",
                "name": "ValidationError"
            },
            {
                "path": [ "stages", "2", "name"],
                "message": "Stage must be unique",
                "name": "ValidationError"
            },
        ]
    }
}
```

What I see in the frontend now is the `Stage must be unique` error displayed 3 times like so:
![Uploading image.png…]()


Do you think this could be inconvenient to handle? Also, is that error format what you were expecting?

Reply:
for future reference @jhoward1994 , we could add something like this in yup

```js
yup.addMethod(yup.array, 'uniqueProperty', function(propertyPath, message) {
    return this.test('unique', '', function(list) {
		// .. Uniqness logic
    });
});
```

So then in any validation it can be used like:
```js
const validateWorkflowUpdateSchema = yup.object().shape({
  name: yup.string().max(255),
  stages: yup
    .array()
    .of(stageObject)
    .uniqueProperty('name', 'This stage must be unique'),
  contentTypes: validateContentTypes,
```

Reply:
> Do you think this could be inconvenient to handle? Also, is that error format what you were expecting?

The format looks good to me. I'll check why it is printing each message instead of the summary, because that is what is expected to happen 👍🏼

Reply:
Done in 3ad06ee3231784e7aae7d1c4f5f40f5b7785bde1

Reply:
Fixed the issue where it was not being validatd in workflow create.
@jhoward1994 I actually added the `uniqueProperty` method in yup, we could use it for the entity-validator.

Reply:
![image](https://github.com/strapi/strapi/assets/20578351/ff6cb497-a480-4ac5-afe9-30d0c03978c5)
Looks great! ✨